### PR TITLE
create `PropertyReference.Psi` from primary constructor properties

### LINKED ISSUE #609: primary constructor properties are not included in `ClassReference.Psi.properties`
Given something like:

```kotlin
class Subject(
  val name: String
)
```

A `ClassReference.Descriptor` instance will include `name` as a property, but the `ClassReference.Psi` version will not.  `Descriptor`'s version is correct.

### PR DESCRIPTION
**Pull request by RBusarow**
fixes #609

### CODE REVIEW

File: compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+      )
+    ) {
+      assertThat(exitCode).isEqualTo(OK)
+    }
+  }
+
+  fun ClassReference.propertyTypeNames() = properties
+    .map { "${it.name}: ${it.type().asTypeName()}" }
+
+  fun List<PropertyReference>.named(name: String) = single { it.name == name }
```
Comment (vRallev) [inline compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt:618]: These functions should be private.

File: compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+              .inOrder()
+          }
+
+          null
+        }
+      )
+    ) {
+      assertThat(exitCode).isEqualTo(OK)
+    }
+  }
```
Comment (vRallev) [inline compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt:613]: Thanks for all these tests!

**Review by vRallev — APPROVED** [review]

File: compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt (hunk 1 above)
Comment (RBusarow) [reply compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt]: fixed

File: compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt (hunk 2 above)
Comment (RBusarow) [reply compiler/src/test/java/com/squareup/anvil/compiler/internal/reference/PropertyReferenceTest.kt:613]: Well, the tooling makes it so easy!

**Pull request merged**